src/core/mac/sub_mac.cpp

```diff
@@ -283,7 +283,8 @@ void SubMac::HandleReceiveDone(RxFrame *aFrame, Error aError)
         mPcapCallback.Invoke(aFrame, false);
     }
 
-    if (!ShouldHandleTransmitSecurity() && aFrame != nullptr && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck)
+    if (!ShouldHandleTransmitSecurity() && aFrame != nullptr && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck &&
+        (aFrame->mInfo.mRxInfo.mAckKeyId == mKeyId))
     {
         SignalFrameCounterUsed(aFrame->mInfo.mRxInfo.mAckFrameCounter);
```

Comment:
Would it be safer to update `SignalFrameCounterUsed()` to have Key ID as a parameter?

Reply:
I considered that as well. It moves the check for `KeyId` from `SubMac` to `KeyManager`. 

Both ways are ok. I went for this model doing it `SuMac` (keep the change in one source file so it may be easier to cherry-pick).

Reply:
Thinking more on this, I agree it may be safer (if this callback in used from other code paths in future) to change it to include key id as a parm. Let me try it.

Reply:
Changed it in new push. We just change the `SubMac::SignalFrameCounterUsed()` to get `aKeyId` and check for a match against `mKeyId` in this method.  No changes to the callback to `KeyManager`.